### DISCUSSION TITLE
Basic setup tutorial (sunny case)

### DISCUSSION THREAD

**Discussion by hilsonp** [vrtmrz/obsidian-livesync#193] [Documentation requests]
I would find interesting for new comers to find a HowTo that would show how to setup from scratch a replicated vault based on:
- One self hosted CouchDB
- One Windows 10 computer
- One OSX computer
- One iOS mobile
- One Android mobile

They could recognise themselves in it, skipping the parts covering devices they don't own (example: I would skip the Android part).

**Comment by dlphnkck** [comment]
Hello. How about either of the following steps?

- Quick setup https://github.com/vrtmrz/obsidian-livesync/blob/main/docs/quick_setup.md
- YouTube https://www.youtube.com/watch?v=HBijC8yVjAk

**Comment by hilsonp** [comment]
I appreciate your support although if I remember well, it was not so straight forward when I did set it up.
@vrtmrz [asked for requests](https://github.com/vrtmrz/obsidian-livesync/discussions/185) to help him prioritise so... here it is.

**Comment by vrtmrz** [comment]
@hilsonp Thank you for the request! Definitely right. This is the dawn.
I rewrote it many times but it wasn't good enough, so I wrote it in a structured way.

- [Setup Self-hosted LiveSync to the first device](https://docs.vrtmrz.net/LiveSync/procedures/Setup+Self-hosted+LiveSync+to+the+first+device)
- [Setup Self-hosted LiveSync to secondary or subsequent devices](https://docs.vrtmrz.net/LiveSync/procedures/Setup+Self-hosted+LiveSync+to+secondary+or+subsequent+devices)

@dlphnkck Thank you for the information! I completely forgot the video.

**Reply by hilsonp** [reply]
Thank you

**Comment by xieve** [comment]
I just set this up, and it was a bit confusing that the tutorial didn't mention whether it was intended for a testing/local setup or a deployment that is visible to the web. It would be great to include best practices for the latter case (which I assume are reverse proxy for SSL, subfolder against crawlers, non-default username—not `admin` or `root` etc., strong password). The fact that you can access the CouchDB URL with a browser for testing should be a bit more highlighted. Further, the config snippets should be explained a bit more: that 5984 is the default CouchDB port, that the config allows for outside connection by default (even though most people are hopefully running a firewall, this might be undesirable! It is certainly not necessary for the reverse proxy setup) and that the config sets HTTP basic auth from the environment, which we then set in the compose file or the run command. 

I also had to disable LiveSync and manually replicate once before anything worked. If that is intended, please document that.

Another important detail missing in the docs is that the CouchDB data volume has to be owned by 1000:1000.

If I didn't miss it, there is no documentation on how to set up the second instance, you just kind of have to figure out how to use the setup URIs.
